packages/waku/src/router/create-pages.ts

```diff
@@ -755,29 +771,39 @@ export const new_createPages = <
     renderRoute: async (path) => {
       await configure();
 
+      // path without slugs
+      const routePath = getRoutePath(path);
+      if (!routePath) {
+        throw new Error('Route not found: ' + path);
+      }
+
       const pageComponent = (staticComponentMap.get(
-        joinPath(path, 'page').slice(1), // feels like a hack
-      ) ?? dynamicPagePathMap.get(path)?.[1])!;
+        joinPath(routePath, 'page').slice(1), // feels like a hack
+      ) ?? dynamicPagePathMap.get(routePath)?.[1])!;
+
+      const pathSpec = parsePathWithSlug(routePath);
+      const mapping = getPathMapping(pathSpec, path);
 
       const result: Record<string, ReactNode> = {
         root: createElement(
           rootItem ? rootItem.component : DefaultRoot,
           null,
           createElement(Children),
         ),
-        [`page:${path}`]: createElement(
+        [`page:${routePath}`]: createElement(
           pageComponent,
-          null,
+          mapping,
```

Comment:
@dai-shi should `query` and `hash` also be sent in here?

Reply:
If a page is dynamic, I think it should receive `query`.

Reply:
thanks, and `hash` only exists on the client, so it is handled inside `NewRouter`? is that right?

Reply:
so there's a bug upstream of new_createPages. I think it's upstream of defineRouter too. 

This prevents the use of query here. I'll open a new PR to talk through a fix for that, but this PR seems like a fine change on its own to me

Reply:
#1004 will fix the `query`

Reply:
thanks. yeah, `hash` is client only.